### PR TITLE
Fix patches!

### PR DESCRIPTION
**Pull request by rainwoodman**
index

### CODE REVIEW

**Comment by pmelchior** [comment]
That looks a lot better. Do you understand why some patches are lost on one side?

**Comment by rainwoodman** [comment]
I believe it is related to I didn't duplicate those vertices crossing the boundary. Ideally we can create a path with two disjoint regions, one per side, -- but I am not sure if matplotlib will be able to correctly fill them -- it probably should.

The latest commit adds two convenient functions: histmap and mapshow to the axes object:

a.histmap(ra, dec, weights=FC, nside=64, mean=True, lw=0, edgecolor='none')

**Comment by rainwoodman** [comment]
or for an existing map:

a.mapshow(m, mask, ....)
where mask decides whether to show the pixel or not.

**Comment by pmelchior** [comment]
OK, great. Two requests:
1. Move the functionality for histmap and mapshow into a free function, with a thin wrapper for the axes class method.
2. I'm a bit puzzled about mapshow. See my note there.

File: aea_projection.py [hunk 1]
```diff
@@ -337,6 +338,28 @@ def set_ylim(self, *args, **kwargs):
...
+            w = r
+        if mean:
+            mask = N != 0
+            w[mask] /= N[mask]
+        else:
+            mask = w > 0
+
+        return w, mask, self.mapshow(w, mask, nest=False, **kwargs)
+
+    def mapshow(self, map, mask, nest=False, **kwargs):
```
Comment (pmelchior) [inline aea_projection.py:368]: this method does not seem to be using `map` at all !? I might be missing something, but that doesn't look right

Comment (rainwoodman) [inline aea_projection.py:368]:
```
coll = PolyCollection(v, array=map[mask], transform=self.transData, **kwargs)
~~~~~~~~~~~~~~~~~~~~~~~~~~ ++++++~~~~~~~~~~
```

**Comment by rainwoodman** [comment]
I am not so sure about '1'. 

These two function are already relatively thin. The real logic is in histogrammap() and _boundary(). 
Is it the idea we will add the other projection and reuse them? Maybe we want to have them share the same base class instead.

File: aea_projection.py (hunk 1 above)
Comment (pmelchior) [inline aea_projection.py:368]: I was blind, sorry.

**Comment by pmelchior** [comment]
I think it's a distinct possibility that we want to another projection. I've done the Lambert Conformal conic already in my old scheme, and the changes are minimal compared to the AEA.
So, I think it would be good to avoid duplication of logic that does not depend on specifics of the transformation.

Having a base class for them would be useful. Short of that, you can move the actual logic out into a free method as you have already done with `histogrammap()` and `_boundary()`.

**Pull request merged**